AlgorithmPlayground/BinaryTreeNode.cs

```diff
@@ -33,21 +33,32 @@ private BinaryTreeNode ConvertArrayToTree(int[] arr, int start, int end)
 			head.Right = ConvertArrayToTree(arr, headpos + 1, end);
 			return head;
 		}
+
 		public void InOrderPrint()
 		{
-			InOrderPrint(this);
+			foreach(BinaryTreeNode n in this.GetInOrder())
+				Console.Write(n.Val + " ");
 			Console.WriteLine();
 		}
-		private void InOrderPrint(BinaryTreeNode n)
+
+		public LinkedList<BinaryTreeNode> GetInOrder()
+		{
+			LinkedList<BinaryTreeNode> results = new LinkedList<BinaryTreeNode>();
+			GetInOrder(results);
+			return results;
+		}
+		private void GetInOrder(LinkedList<BinaryTreeNode> results)
```

Comment:
This could probably yield the results one by one which would be cool for really big trees